src/main/java/build/buildfarm/common/net/URL.java

```diff
@@ -16,16 +16,28 @@
 
 import java.io.IOException;
 import java.net.URLConnection;
+import java.util.Optional;
 
-// solely exists so that we can mock this
 public class URL {
   private final java.net.URL url;
+  private final Optional<String> authHeader;
 
+  public URL(java.net.URL url, String authHeader) {
```

Comment:
So i noticed the comment on line 20, but since this class already had the `openConnection` method I thought hey, this class is already it's own person. So I removed the comment instead. Might need a new name though.